ContextDesktop/src/main/java/com/github/migi_1/Context/enemy/EnemyMoveBehaviour.java

```diff
@@ -0,0 +1,176 @@
+package com.github.migi_1.Context.enemy;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Random;
+
+import com.github.migi_1.Context.model.entity.Carrier;
+import com.github.migi_1.Context.model.entity.EnemySpot;
+import com.github.migi_1.Context.model.entity.behaviour.MoveBehaviour;
+import com.jme3.math.Vector3f;
+
+/**
+ * Class responsible for the behaviour of enemies.
+ * @author Damian
+ *
+ */
+public class EnemyMoveBehaviour extends MoveBehaviour {
+
+    private static final float DISTANCE_THRESHOLD = 50;
+    private Vector3f moveVector;
+    private static final float STARTING_SPEED = 0.5f;
+    private Vector3f localTranslation;
+    private ArrayList<Carrier> carriers;
```

Comment:
Since an enemy has only one carrier as its target, it might be cleaner to make a field only for this carrier and delete the field with the list of carriers. I would recommend passing the list of carriers to the createTargetSpot method and initialising the target carrier there.